Document the new notifications

Hunterness opened the pull request:
returned by `CREATE INDEX ... IF NOT EXISTS ...`, `CREATE CONSTRAINT ... IF NOT EXISTS ...`, `DROP INDEX ... IF EXISTS`, and `DROP CONSTRAINT ... IF EXISTS` in the cases when nothing happens.

Hunterness committed:
Document the new notifications

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-880.surge.sh

Updated pages:
- [Examples](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/examples)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/syntax)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Create, show, and delete indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/search-performance-indexes/managing-indexes)
- [Full-text indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/full-text-indexes)
- [Vector search indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/vector-indexes)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/syntax)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

HannesSandberg reviewed (approved):
Looks good to me 👍

Hunterness commented on modules/ROOT/pages/constraints/examples.adoc:
```diff
@@ -472,6 +472,12 @@ Assuming a constraint with the name `sequels` already exists:
 (no changes, no records)
 ----
 
+.Notification
```
For the docs team review: I'm not sure how we want to represent notifications and if/how we want to add it to the docs testing

rsill-neo4j replied:
thanks, i'll ask in our daily stand-up tomorrow

stefano-ottolenghi replied:
I think the choice you've made is good. There's no plan to test notifications as of now, also given how few of them we have in docs (if any, beyond this addition).

rsill-neo4j commented on modules/ROOT/pages/constraints/examples.adoc:
```diff
@@ -99,7 +99,7 @@ It will be updated to say `Node property uniqueness constraints added: 1` in Neo
 To avoid such an error, `IF NOT EXISTS` can be added to the `CREATE` command.
 This will ensure that no error is thrown and that no constraint is created if any other constraint with the given name, or another node property uniqueness constraint on the same schema, already exists.
-
+Instead, an informational notification will be returned showing the pre-existing constraint blocking the creation.
```
```suggestion
label:new[Introduced in 5.17] An informational notification is instead returned showing the existing constraint which blocks the creation.
```

multiple occurrences

edit: added label

rsill-neo4j commented on modules/ROOT/pages/constraints/syntax.adoc:
```diff
@@ -16,6 +16,7 @@ This means its default behavior is to throw an error if an attempt is made to cr
 With the `IF NOT EXISTS` flag, no error is thrown and nothing happens should a constraint with the same name or same schema and constraint type already exist.
 It may still throw an error if conflicting data, indexes, or constraints exist.
 Examples of this are nodes with missing properties, indexes with the same name, or constraints with same schema but a different conflicting constraint type.
+In the case when nothing happens an informational notification will be returned showing the pre-existing constraint blocking the creation.
```
```suggestion
label:new[Introduced in 5.17] In case nothing happens an informational notification is returned showing the existing constraint which blocks the creation.
```

multiple occurrences

edit: added label

rsill-neo4j commented on modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc:
```diff
@@ -19,6 +19,58 @@ Replacement syntax for deprecated and removed features are also indicated.
+----
+|
+When attempting to create an index using `IF NOT EXISTS` with either the same name or same index type and schema, or both,
+as an existing index the command will now return a notification showing the pre-existing index blocking the creation.
```
```suggestion
as an existing index the command now returns a notification showing the existing index which blocks the creation.
```

rsill-neo4j commented on modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc:
```diff
@@ -19,6 +19,58 @@ Replacement syntax for deprecated and removed features are also indicated.
+----
+|
+When attempting to create a constraint using `IF NOT EXISTS` with either the same name or same constraint type and schema
+(and property type for property type constraints), or both, as an existing constraint the command will now return
+a notification showing the pre-existing constraint blocking the creation.
```
```suggestion
(and property type for property type constraints), or both, as an existing constraint the command now returns
a notification showing the existing constraint which blocks the creation.
```

rsill-neo4j commented on modules/ROOT/pages/indexes/search-performance-indexes/managing-indexes.adoc:
```diff
@@ -1009,6 +1038,7 @@ DROP INDEX index_name [IF EXISTS]
 The `DROP INDEX` command is optionally idempotent.
 This means that its default behavior is to throw an error if an attempt is made to drop the same index twice.
 With `IF EXISTS`, no error is thrown and nothing happens should the index not exist.
+Instead, an informational notification will be returned detailing that the index does not exist.
```
```suggestion
label:new[Introduced in 5.17] Instead, an informational notification is returned detailing that the index does not exist.
```

multiple occurrences

edit: added label

rsill-neo4j commented on modules/ROOT/pages/indexes/search-performance-indexes/managing-indexes.adoc:
```diff
@@ -1096,4 +1126,9 @@ The following statement will attempt to drop the index named `missing_index_name
 ----
 
-If an index with that name exists it is removed, if not the command does nothing.
+If an index with that name exists it is removed, if not the command does nothing and an informational notification is returned.
```
```suggestion
If an index with that name exists it is removed, if not the command does nothing.
label:new[Introduced in 5.17] Additionally, an informational notification is returned.
```

edit: added label

stefano-ottolenghi reviewed (commented):
A couple comments, even if I appreciate not all the issues come strictly from this PR:

1. `Node property uniqueness constraints > Handling existing constraints when creating a constraint` contains the info sentence but the example was not tweaked to include the notification, as the others were.
2. The sentences like "Assuming a constraint with the name `sequels` already exists:" are misleading, as the constraint _name_ doesn't matter. Rather, it should say "In case a constraint on `sequel.order` already exists:". There's at least a couple instances like this.

2. if you wish the `Result` and `Notification` blocks to be uniform in style with the example blocks (at least for how the styles are laid down right now), they should all be `source` blocks: 
```adoc
.Result
[source, queryresult]
----
(no changes, no records)
----

.Notification
[source]
----
`CREATE CONSTRAINT sequels IF NOT EXISTS FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order) IS UNIQUE` has no effect.
`CONSTRAINT sequels FOR ()-[e:SEQUEL_OF]-() REQUIRE (e.order, e.seriesTitle) IS UNIQUE` already exists.
----
```

Hunterness commented:
@stefano-ottolenghi 

1. That example actually creates a new constraint, so there is no notification.
2. That depends, we check for either the label/rel type and property OR the name, and I think some of the examples have different label/reltype/properties and only  share the name and that it's those that have that version of the sentence
3. good point, I didn't really know what to put there so I started with nothing XD

Hunterness committed:
Review comments

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-880.surge.sh

Updated pages:
- [Examples](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/examples)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/syntax)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Create, show, and delete indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/search-performance-indexes/managing-indexes)
- [Full-text indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/full-text-indexes)
- [Vector search indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/vector-indexes)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/syntax)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

Hunterness committed:
Review comments part 2

Added the label instead (and the 5.16 version for parameter name so that they match (as it was the sentences after each other in the fulltext chapter, and it looked odd for them to be different))

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-880.surge.sh

Updated pages:
- [Examples](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/examples)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/syntax)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Create, show, and delete indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/search-performance-indexes/managing-indexes)
- [Full-text indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/full-text-indexes)
- [Vector search indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/vector-indexes)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/syntax)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

neo-technology-commit-status-publisher commented:
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-880.surge.sh

Updated pages:
- [Examples](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/examples)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/constraints/syntax)
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)
- [Create, show, and delete indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/search-performance-indexes/managing-indexes)
- [Full-text indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/full-text-indexes)
- [Vector search indexes](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/semantic-indexes/vector-indexes)
- [Syntax](https://neo4j-docs-cypher-880.surge.sh/cypher-manual/5/indexes/syntax)
<!-- Sticky Pull Request Commentdocs-pr-changes -->

stefano-ottolenghi commented on modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc:
```diff
@@ -19,6 +19,58 @@ Replacement syntax for deprecated and removed features are also indicated.
+----
+|
+When attempting to create an index using `IF NOT EXISTS` with either the same name or same index type and schema, or both,
+as an existing index the command now returns a notification showing the existing index which blocks the creation.
```
Please don't break text in-between sentences. Either break it after full stops, or not at all. There's multiple occurrences in this file. After addressing this, it's good to merge as far as I can see :cartwheeling:

Hunterness replied:
I did break them while writing to be able to see the sentences since they were so long (as I don't have soft wrapping so the lines just continue out of sight and requires scrolling to see the end) but I can definitely go back and fix that 👍

Hunterness committed:
Review comments part 3

Don't split sentences over multiple lines

neo-technology-commit-status-publisher commented:
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.
<!-- Sticky Pull Request Commentdocs-pr-changes -->
